compiler/hash-types/src/new/symbols.rs

```diff
@@ -0,0 +1,15 @@
+//! Definitions related to symbols and names.
+
+use hash_source::identifier::Identifier;
+use hash_utils::{new_store_key, store::DefaultStore};
+
+/// The data carried by a symbol.
+///
+/// This is basically just a name identifier.
+///
+/// This is used to avoid needing to perform alpha-conversion on terms.
+pub struct SymbolData {
+    pub name: Identifier,
+}
```

Comment:
Will there be more information that is added to this `SymbolData`?

Reply:
Not really, at least not as far as I can see now. Auxiliary information like locations might be stored in other maps.

Reply:
So, why not use `Identifier` directly?

Reply:
Two same identifiers might have different symbols. For example `a : = 3; { a := 4; }` the inner `a` will have a different `SymbolId` to the outer `a`.